js/packages/components/chat/footer/recorder/RecorderButton.tsx

```diff
@@ -145,21 +128,30 @@ export const RecorderButton: React.FC<RecorderButtonProps> = ({
 	}, [addMeteredValue, recorder, setRecorderFilePath])
 
 	const handlePressRecording = useCallback(async () => {
-		const permState = await acquireMicPerm(navigate)
-		if (permState !== MicPermStatus.GRANTED) {
-			setHelpMessageValue({ message: t('audio.record.tooltip.permission-denied') }) //'App is not allowed to record sound'
-			return
-		}
+		const status = (await getPermissions()).audio
 
-		clearHelpMessageValue()
-		setRecordingStart(Date.now())
-		setCurrentTime(Date.now())
-		setClearRecordingInterval(setInterval(() => setCurrentTime(Date.now()), 100))
-		meteredValuesRef.current = []
+		await checkPermission({
+			permissionType: PermissionType.audio,
+			navigate,
+			accept: () => {
+				// We check the status before ask it, to don't record after asking permission
+				// (In other apps, the recorder don't start after accept permission)
+				if (status === RESULTS.GRANTED) {
```

Comment:
I think `status` is out of date here

Reply:
What do you mean by out of date ?

Reply:
I mean that by the time the `accept` callback is called, the status value might be different

Reply:
Mhhh, but this value is usefull just at the init

Reply:
But i want the `status` value before call `accept` callback

Reply:
I can change his name to be more clear like `initialStatus` something like that